File: Assistant/LifecycleEvents/app/src/main/java/id/ac/unsyiah/lifecycleevents/MainActivity.java
package id.ac.unsyiah.lifecycleevents;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    public static final String MAIN_ACTIVITY = "MainActivity";
    public static final String LOG_TAG = MAIN_ACTIVITY;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(LOG_TAG, "onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(LOG_TAG, "onStart");
    }
}

File: skillacademy/NavigationComponent/app/src/main/java/com/bnawan/navigationcomponent/MainActivity.kt
package com.bnawan.navigationcomponent

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.bnawan.navigationcomponent.model.User
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val user = User(name="Budi Gunawan", age=32)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_open.setOnClickListener { toDetail() }
    }

    private fun toDetail() {
        startActivity(
            Intent(this,DetailActivity::class.java)
                .apply { putExtra(User::class.java.simpleName, user) }
        )
    }
}
